From 12c699d9350fe643b3b0950180689835608f580f Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:24 +0100
Subject: [PATCH 30/56] media: intel/ipu4: Update set_routing callback to match
 v4l2_subdev_pad_ops signature

Adapt the set_routing callback implementation to match the updated v4l2_subdev_pad_ops
signature, which now includes state and format whence parameters.
---
 drivers/media/pci/intel/ipu-isys-subdev.c | 4 +++-
 drivers/media/pci/intel/ipu-isys-subdev.h | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-subdev.c b/drivers/media/pci/intel/ipu-isys-subdev.c
index e5c93f486976..ab7500529abe 100644
--- a/drivers/media/pci/intel/ipu-isys-subdev.c
+++ b/drivers/media/pci/intel/ipu-isys-subdev.c
@@ -496,7 +496,9 @@ bool ipu_isys_subdev_has_route(struct media_entity *entity,
 }
 
 int ipu_isys_subdev_set_routing(struct v4l2_subdev *sd,
-				struct v4l2_subdev_routing *route)
+			   struct v4l2_subdev_state *state,
+			   enum v4l2_subdev_format_whence which,
+			   struct v4l2_subdev_krouting *route)
 {
 	struct ipu_isys_subdev *asd = to_ipu_isys_subdev(sd);
 	int i, j, ret = 0;
diff --git a/drivers/media/pci/intel/ipu-isys-subdev.h b/drivers/media/pci/intel/ipu-isys-subdev.h
index e59f846550e9..1cbf82858bf5 100644
--- a/drivers/media/pci/intel/ipu-isys-subdev.h
+++ b/drivers/media/pci/intel/ipu-isys-subdev.h
@@ -165,7 +165,9 @@ void ipu_isys_subdev_cleanup(struct ipu_isys_subdev *asd);
 int ipu_isys_subdev_get_frame_desc(struct v4l2_subdev *sd,
 				   struct v4l2_mbus_frame_desc *desc);
 int ipu_isys_subdev_set_routing(struct v4l2_subdev *sd,
-				struct v4l2_subdev_routing *route);
+			   struct v4l2_subdev_state *state,
+			   enum v4l2_subdev_format_whence which,
+			   struct v4l2_subdev_krouting *route);
 int ipu_isys_subdev_get_routing(struct v4l2_subdev *sd,
 				struct v4l2_subdev_routing *route);
 bool ipu_isys_subdev_has_route(struct media_entity *entity,
-- 
2.51.0

